crates/hir_ty/src/display.rs

```diff
@@ -1154,7 +1154,7 @@ impl HirDisplay for Path {
         }
 
         for (seg_idx, segment) in self.segments().iter().enumerate() {
-            if seg_idx != 0 {
+            if seg_idx != 0 || matches!(self.kind(), PathKind::Crate) {
```

Comment:
it seems like `super` and `self` would have the same problems?

Reply:
I knew, I was just fixing it. It should be okay for now.

I'll add some tests later.